src/Compilers/CSharp/Portable/Errors/MessageID.cs

```diff
@@ -305,6 +306,7 @@ internal static LanguageVersion RequiredVersion(this MessageID feature)
                 case MessageID.IDS_FeatureTargetTypedObjectCreation: // syntax check
                 case MessageID.IDS_FeatureMemberNotNull:
                 case MessageID.IDS_FeatureNativeInt:
+                case MessageID.IDS_FeatureModuleInitializers:
```

Comment:
>case MessageID.IDS_FeatureModuleInitializers: [](start = 16, length = 45)

I assume this is not a syntax check, consider adding a comment.

Reply:
If IDS_FeatureModuleInitializers is not a syntax check, I don't believe `IDS_FeatureMemberNotNull` is a syntax check. Is that where the comment would go?

Reply:
> If IDS_FeatureModuleInitializers is not a syntax check, I don't believe IDS_FeatureMemberNotNull is a syntax check. Is that line where the comment would go? 

For module initializers the comment would go at the end of ```case MessageID.IDS_FeatureModuleInitializers:``` line.

> Something like // not a syntax check?

We usually say ```// semantic check```, but it would be fine to be even more specific. Something like checked when such-and-such well-known attribute is encountered on a method, for example. 

---
In reply to: [409483403](https://github.com/dotnet/roslyn/pull/43281#discussion_r409483403) [](ancestors = 409483403)